### PR TITLE
Victor K data-structures-stacks 1st submission

### PR DESCRIPTION
**Pull request by vkalinkin**
screencapture-file-C-Users-vakal-repos-rc0521-code-solutions-data-structures-stacks-index-html-2021-08-03-09_27_27

### CODE REVIEW

File: data-structures-stacks/count-values.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* exported countValues */
+
+function countValues(stack) {
+  let count = 0;
+  let topValue;
+  topValue = stack.peek();
+  if (topValue === undefined) {
+    return count;
+  } else {
```
Comment (scbowler) [inline data-structures-stacks/count-values.js:9]: You don't really need this extra check, your while loop would just never run if `topValue` is `undefined`

File: data-structures-stacks/count-values.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    while (topValue !== undefined) {
+      stack.pop();
+      topValue = stack.peek();
```
Comment (scbowler) [inline data-structures-stacks/count-values.js:12]: You can just use `pop` here and save the result in `topValue`

File: data-structures-stacks/count-values.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* exported countValues */
+
+function countValues(stack) {
+  let count = 0;
+  let topValue;
+  topValue = stack.peek();
```
Comment (scbowler) [inline data-structures-stacks/count-values.js:6]: Use `pop` here as well

File: data-structures-stacks/max-value.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/* exported maxValue */
+
+function maxValue(stack) {
+  let maxValue;
```
Comment (scbowler) [inline data-structures-stacks/max-value.js:4]: If you start `maxValue` as `-Infinity` it is helpful and can save you some checks below

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good! A few optimizations requested. Let us know if you have any questions!

**Comment by vkalinkin** [comment]
stack count values
stack max value

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍 

Remember to remove commented out code 😉

**Pull request merged**